modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/trait/WithEventBus.java

```diff
@@ -0,0 +1,158 @@
+package de.wuespace.telestion.api.verticle.trait;
+
+import de.wuespace.telestion.api.message.JsonMessage;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Verticle;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Allows {@link Verticle} instances to get simplified access to the Vert.x event bus.
+ * All methods allow usage with {@link JsonMessage} messages, too.
+ *
+ * <h2>Usage</h2>
+ * <pre>
+ * {@code
+ * public class MyVerticle extends TelestionVerticle implements WithEventBus {
+ *     @Override
+ *     public void startVerticle() {
+ *         register("channel-1", this::handle, SimpleMessage.class);
+ *     }
+ *
+ *     private void handle(SimpleMessage body) {
+ *         logger.info("Received Telecommand: {}", body.tcClass());
+ *     }
+ * }
+ * }
+ * </pre>
+ *
+ * @author Pablo Klaschka, Ludwig Richter
+ */
+public interface WithEventBus extends Verticle {
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#publish(String, Object, DeliveryOptions)
+	 */
+	default void publish(String address, JsonObject message, DeliveryOptions options) {
+		getVertx().eventBus().publish(address, message, options);
+	}
+
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#publish(String, Object)
+	 */
+	default void publish(String address, JsonObject message) {
+		getVertx().eventBus().publish(address, message);
+	}
+
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#publish(String, Object, DeliveryOptions)
+	 */
+	default void publish(String address, JsonMessage message, DeliveryOptions options) {
+		publish(address, message.json(), options);
+	}
+
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#publish(String, Object)
+	 */
+	default void publish(String address, JsonMessage message) {
+		publish(address, message.json());
+	}
+
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#send(String, Object, DeliveryOptions)
+	 */
+	default void send(String address, JsonObject message, DeliveryOptions options) {
+		getVertx().eventBus().send(address, message, options);
+	}
+
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#send(String, Object)
+	 */
+	default void send(String address, JsonObject message) {
+		getVertx().eventBus().send(address, message);
+	}
+
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#send(String, Object, DeliveryOptions)
+	 */
+	default void send(String address, JsonMessage message, DeliveryOptions options) {
+		send(address, message.json(), options);
+	}
+
+	/**
+	 * @see io.vertx.core.eventbus.EventBus#send(String, Object)
+	 */
+	default void send(String address, JsonMessage message) {
+		send(address, message.json());
+	}
```

Comment:
Should there be a `default void rawSend(String address, Object obj)` so it matches the raw implementation of `EventBus.send()` to only extend the functionality? Otherwise you cannot send "raw" integer values for instance...

Same with publish() and requst()

Reply:
Good point!

Reply:
Done in https://github.com/wuespace/telestion-core/commit/6427b7ccc9e9474ca5562f7b5bf40bf101adc2f4 :wink: